nav2_tasks/include/nav2_tasks/bt_action_node.hpp

```diff
@@ -69,13 +91,15 @@ class BtActionNode : public BT::CoroActionNode
 
       switch (status) {
         case nav2_tasks::TaskStatus::SUCCEEDED:
+          setStatus(BT::NodeStatus::IDLE);
           return BT::NodeStatus::SUCCESS;
 
         case nav2_tasks::TaskStatus::FAILED:
+          setStatus(BT::NodeStatus::IDLE);
```

Comment:
Can you explain why you are setting the status to IDLE in every case? I'm not sure I follow the intended logic

Reply:
I wanted to ensure that the node was ready to be used again since the BT will be re-used and not destroyed. It all cases (SUCCEEDED, ABORTED, or CANCELED) the node has run to completion. It can no go into the IDLE state where it is not running, but ready to run again.